docs/src/test/java/docs/user/effects/ShoppingCartEntity.java

```diff
@@ -0,0 +1,58 @@
+package docs.user.effects;
+
+import com.example.Hotitems;
+import com.example.Shoppingcart;
+import com.google.protobuf.Empty;
+import io.cloudstate.javasupport.Context;
+import io.cloudstate.javasupport.ServiceCallRef;
+import io.cloudstate.javasupport.eventsourced.CommandContext;
+import io.cloudstate.javasupport.eventsourced.CommandHandler;
+
+public class ShoppingCartEntity {
+
+  // #lookup
+  private final ServiceCallRef<Hotitems.Item> itemAddedToCartRef;
+
+  public ShoppingCartEntity(Context ctx) {
+    itemAddedToCartRef =
+        ctx.serviceCallFactory()
+            .lookup("example.shoppingcart", "ItemAddedToCart", Hotitems.Item.class);
+  }
+  // #lookup
+
+  class CommandHandlerWithForward {
+    // #forward
+    @CommandHandler
+    public void addItem(Shoppingcart.AddLineItem item, CommandContext ctx) {
+      // ... Validate and emit event
+
+      ctx.forward(
+          itemAddedToCartRef.createCall(
```

Comment:
An idea w.r.t. naming. How about `prepareCall`?

Reply:
Agree it's not the best naming, but what exactly is being prepared? We're creating something that has a reference to a call and the message to pass to it - prepare sounds like something more than that is being done, eg perhaps resources are being allocated ready to make the call, which is not the case.